src/Illuminate/Database/Eloquent/FactoryBuilder.php

```diff
@@ -131,7 +132,22 @@ protected function makeInstance(array $attributes = [])
 
             $definition = call_user_func($this->definitions[$this->class][$this->name], $this->faker, $attributes);
 
-            return new $this->class(array_merge($definition, $attributes));
+            $evaluated = $this->evaluateClosures(array_merge($definition, $attributes));
+
+            return new $this->class($evaluated);
         });
     }
+
+    /**
+     * Delay closures evaluation after merging to avoid duplication.
+     *
+     * @param  array  $attributes
+     * @return array
+     */
+    protected function evaluateClosures(array $attributes)
+    {
+        return array_map(function ($attribute) use ($attributes) {
+            return $attribute instanceof Closure ? $attribute($attributes) : $attribute;
+        }, $attributes);
```

Comment:
@JosephSilber You're right. But for what I know at the moment there are no tests at all on the FactoryBuilder class and as this is my first PR to Laravel I would wait for some input before implementing some.